app/forms/supplier_user_form.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class SupplierUserForm
+  include ActiveModel::Model
+  include ActiveModel::Serialization
+
+  attr_accessor :full_name, :email, :supplier
+  validates :supplier, presence: { message: "Select one" }, on: :supplier
+  validates :full_name, presence: { message: "Enter a name" }, on: :details
+  validates :email, presence: { message: "Enter email" }, on: :details
+  validate :email_not_taken, on: :details
+
+  def attributes
+    { "full_name" => nil, "email" => nil, "supplier" => nil }
+  end
+
+  def chosen_supplier
+    LeadProvider.find_by(id: supplier) || DeliveryPartner.find_by(id: supplier)
+  end
+
+  def save!
+    profile = if chosen_supplier.instance_of?(LeadProvider)
+                LeadProviderProfile.new(lead_provider: chosen_supplier)
+              else
+                DeliveryPartnerProfile.new(delivery_partner: chosen_supplier)
+              end
```

Comment:
Ech, I really dislike that we manage LeadProviders and DeliveryPartners in the same places and in the same forms :(

But I guess it's what our designers want.

Reply:
🤷 wireframes, innit